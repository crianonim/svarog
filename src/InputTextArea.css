.parse-input-ta {
    width: 100%;
    height: 5rem;
  }