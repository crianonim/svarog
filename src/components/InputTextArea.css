.parse-input-ta {
    height: 5rem;
  }