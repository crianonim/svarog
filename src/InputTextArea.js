import React from "react";
import './InputTextArea.css';

const InputTextArea = (props) => {
    return (
        <>
         <textarea className="parse-input-ta"></textarea>
         <button onClick={()=>{
         let id=1;
         const fakeDOM=document.createElement('div');
         fakeDOM.innerHTML=document.querySelector('textarea').value;
         const svg=fakeDOM.children[0];
         const shapes=Array.from(svg.children).map(child=>{
           const atts=Object.fromEntries(child.getAttributeNames().map(att=>[att,child.getAttribute(att)]));
           return {id:id++,attributes:atts,shape:child.tagName}
        })
        props.change(shapes);
      }
      }>parse</button>
        </>
    )
}

export default InputTextArea;