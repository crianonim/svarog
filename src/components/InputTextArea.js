import React from "react";
import './InputTextArea.css';

const InputTextArea = (props) => {
    return (
      <div className="panel">
        <div className="panel-heading">Parse your SVG code</div>
         <div className="panel-block">  
        <div className="flex-column  margined">
         <textarea className="parse-input-ta"></textarea>
         <button className="button is-small is-info" onClick={()=>{
         let id=1;
         const fakeDOM=document.createElement('div');
         fakeDOM.innerHTML=document.querySelector('textarea').value;
         const svg=fakeDOM.children[0];
         if (svg) {
           const shapes=Array.from(svg.children).map(child=>{
             const atts=Object.fromEntries(child.getAttributeNames().map(att=>[att,child.getAttribute(att)]));
             return {id:id++,attributes:atts,shape:child.tagName}
            })
            props.change(shapes);
          } else {
            props.msg("E#Badly formed SVG.")
          }
      }
      }>Parse</button>
        </div>
        </div>
      
      </div>      
    )
}

export default InputTextArea;