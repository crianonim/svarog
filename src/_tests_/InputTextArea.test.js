import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import InputTextArea from "../components/InputTextArea";

it("matches a snapshrot",()=>{
    const wrapper=shallow(<InputTextArea/>) 
    expect(toJson(wrapper)).toMatchSnapshot();
});
// TODO: implement different way to create element
// it("changes is fired when add property is clicked",()=>{
//     const changed=jest.fn();
//     const msg=jest.fn();
//     // global.document={querySelector:()=>({value:"<svg></svg>"})}
//     const wrapper=shallow(<InputTextArea changed={changed} msg={msg}/>) 
//     wrapper.find("button").simulate('click');
//     expect(changed).toHaveBeenCalledTimes(1);
// })
