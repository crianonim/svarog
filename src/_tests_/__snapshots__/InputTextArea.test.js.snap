// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshrot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-heading"
  >
    Parse your SVG code
  </div>
  <div
    className="panel-block"
  >
    <div
      className="flex-column  margined"
    >
      <textarea
        className="parse-input-ta"
      />
      <button
        className="button is-small is-info"
        onClick={[Function]}
      >
        Parse
      </button>
    </div>
  </div>
</div>
`;
